--- a/manifests/version.pp     2014-08-27 12:24:57.659804760 +0100
+++ b/manifests/version.pp  2014-08-27 12:25:06.841971767 +0100
@@ -5,7 +5,7 @@
 class apache::version {
   case $::osfamily {
     'RedHat': {
-      if ($::operatingsystem == 'Fedora' and $::operatingsystemrelease >= 18) or ($::operatingsystem != 'Fedora' and $::operatingsystemrelease >= 7) {
+      if ($::operatingsystem == 'Fedora' and $::operatingsystemrelease >= 18) or ($::operatingsystem != 'Fedora' and $::operatingsystemmajrelease >= 7) {
         $default = 2.4
       } else {
         $default = 2.2
