diff --git a/manifests/params.pp b/manifests/params.pp
index f553567..e62f58d 100644
--- a/manifests/params.pp
+++ b/manifests/params.pp
@@ -44,8 +44,8 @@
         'Fedora', 'RedHat': {
           $special_service_provider = undef
         }
-        'RedHat': {
-          if ($::operatingsystemrelease < 7) {
+        'RedHat', 'CentOS', 'Scientific': {
+          if ($::operatingsystemmajrelease < 7) {
             $special_service_provider = 'init'
           } else {
             $special_service_provider = undef
