diff --git a/packstack/puppet/templates/amqp.pp b/packstack/puppet/templates/amqp.pp
index 15d7b5d..ca79707 100644
--- a/packstack/puppet/templates/amqp.pp
+++ b/packstack/puppet/templates/amqp.pp
@@ -44,7 +44,7 @@
     }
 
     'RedHat', 'CentOS': {
-      if $::operatingsystemrelease >= 7 {
+      if $::operatingsystemmajrelease >= 7 {
         $config = '/etc/qpid/qpidd.conf'
       } else {
         $config = '/etc/qpidd.conf'
diff --git a/packstack/puppet/templates/mysql_install.pp b/packstack/puppet/templates/mysql_install.pp
index 1364f79..6a33cba 100644
--- a/packstack/puppet/templates/mysql_install.pp
+++ b/packstack/puppet/templates/mysql_install.pp
@@ -1,6 +1,6 @@
 
 # on EL6 we need to wait for innodb changes before starting mysqld
-if $::operatingsystem in ['RedHat','CentOS','Scientific'] and $::operatingsystemrelease < 7 {
+if $::operatingsystem in ['RedHat','CentOS','Scientific'] and $::operatingsystemmajrelease < 7 {
     $manage_service = false
     service { 'mysqld':
       enable  => true,
diff --git a/packstack/puppet/templates/prescript.pp b/packstack/puppet/templates/prescript.pp
index 4c95292..32d53e8 100644
--- a/packstack/puppet/templates/prescript.pp
+++ b/packstack/puppet/templates/prescript.pp
@@ -10,7 +10,7 @@
 }
 
 # For older RHEL-6 releases kernel/iptools does not support netns
-if $::operatingsystem in $el_releases and $::operatingsystemrelease < 7 {
+if $::operatingsystem in $el_releases and $::operatingsystemmajrelease < 7 {
   $info = "The RDO kernel that includes network namespace (netns) support has been installed on host $::ipaddress."
   $warning = " This is a community supplied kernel and is not officially supported by Red Hat. Installing this kernel on RHEL systems may impact your ability to get support from Red Hat."
 
